lib/LWP/UserAgent.pm

```diff
@@ -747,6 +747,23 @@ sub cookie_jar {
     $old;
 }
 
+sub proxy_header {
+    my $self = shift;
+    $self->{proxy_headers} ||= HTTP::Headers->new;
+    return $self->{proxy_headers}->header(@_);
+}
+
+sub proxy_headers {
+    my $self = shift;
+    my $old = $self->{proxy_headers} ||= HTTP::Headers->new;
+    if (@_) {
+	Carp::croak("proxy_headers not set to HTTP::Headers compatible object")
+	    unless @_ == 1 && $_[0]->can("header_field_names");
```

Comment:
Yes, I'm just not sure the original is correct in this case.  `->can()` assumes that an object has been passed.  If something else is passed to this method then we'll lose the intended error message.  Something like `unless @_ == 1 && blessed( $_[0] ) && $_[0]->can('header_field_names')` would cover that case.